src/components/Toast/toastStore.ts

```diff
@@ -0,0 +1,92 @@
+import { nanoid } from 'nanoid';
+import { action, atom, WritableAtom } from 'nanostores';
+import { ToastType } from './enums';
+import { Toast, ToastOptions, ToastUpdateOptions } from './types';
+
+export const toastStore = atom<Toast[]>([]);
+
+export const updateToast = action(
+  toastStore,
+  'updateToast',
+  (
+    store: WritableAtom<Toast[]>,
+    toastId: Toast['id'],
+    options: ToastUpdateOptions
+  ) => {
+    const existingToasts = store.get();
+    store.set(
+      existingToasts.map((toast) => {
+        if (toast.id === toastId) {
+          return {
+            ...toast,
+            ...options,
+          };
+        } else {
+          return toast;
+        }
+      })
+    );
+  }
+);
+
+const removeToast = action(
+  toastStore,
+  'removeToast',
+  (store: WritableAtom<Toast[]>, toastId: Toast['id']) => {
+    const existingToasts = store.get();
+    store.set(existingToasts.filter((toast) => toast.id !== toastId));
+  }
+);
+
+export const dismissToast = action(
+  toastStore,
+  'dismissToast',
+  (store: WritableAtom<Toast[]>, toastId: Toast['id']) => {
+    const existingToasts = store.get();
+    store.set(
+      existingToasts.map((toast) => {
+        if (toast.id === toastId) {
+          return {
+            ...toast,
+            dismissed: true,
+          };
+        } else {
+          return toast;
+        }
+      })
+    );
+    setTimeout(() => {
+      removeToast(toastId);
+    }, 175);
```

Comment:
Please store numbers like this (magic numbers in general) in variables, thus they will be more descriptive.

Reply:
Oh good catch, thanks!